src/image/transform/mask/yen.js

```diff
@@ -0,0 +1,50 @@
+// see https://github.com/fiji/Auto_Threshold/blob/master/src/main/java/fiji/threshold/Auto_Threshold.java
+// Implements Yen  thresholding method
+// 1) Yen J.C., Chang F.J., and Chang S. (1995) "A New Criterion
+//    for Automatic Multilevel Thresholding" IEEE Trans. on Image
+//    Processing, 4(3): 370-378
+// 2) Sezgin M. and Sankur B. (2004) "Survey over Image Thresholding
+//    Techniques and Quantitative Performance Evaluation" Journal of
+//    Electronic Imaging, 13(1): 146-165
+//    http://citeseer.ist.psu.edu/sezgin04survey.html
+//
+// M. Emre Celebi
+// 06.15.2007
+// Ported to ImageJ plugin by G.Landini from E Celebi's fourier_0.8 routines
+
+export default function yen(histogram, total) {
+    let threshold;
+    let crit;
+    let max_crit;
+    let norm_histo = []; /* normalized histogram */
+    let P1 = []; /* cumulative normalized histogram */
+    let P1_sq = [];
+    let P2_sq = [];
```

Comment:
If you can know in advance the size of an array, it is important to initialize it with the correct size. This can be done with `new Array(size)`.
For example: `let norm_histo = new Array(histogram.length);`